include/picongpu/particles/Particles.hpp

```diff
@@ -72,7 +79,29 @@ class Particles : public ParticlesBase<
         T_Name,
         SuperCellSize,
         T_Attributes,
-        T_Flags
+        T_Flags,
+        typename bmpl::if_<
+            // check if alias boundaryCondition is defined for the species
+            bmpl::contains<
+                T_Flags,
+                typename GetKeyFromAlias<
+                    T_Flags,
+                    boundaryCondition< >
+                >::type
+            >,
+            // resolve the alias
+            typename pmacc::traits::Resolve<
+                typename GetKeyFromAlias<
+                    T_Flags,
+                    boundaryCondition< >
+                >::type
+            >::type,
+            // fallback if the species has not defined the alias boundaryCondition
+            pmacc::HandleGuardRegion<
+                pmacc::particles::policies::ExchangeParticles,
+                GuardHandlerCallPlugins
```

Comment:
please let us rename this class in a follow-up to `CallPluginsAndDeleteParticles` or so